packages/search-in-workspace/src/browser/search-in-workspace-widget.tsx

```diff
@@ -452,11 +463,21 @@ export class SearchInWorkspaceWidget extends BaseWidget implements StatefulWidge
     protected performSearch(): void {
         const searchOptions: SearchInWorkspaceOptions = {
             ...this.searchInWorkspaceOptions,
+            followSymlinks: this.shouldFollowSymlinks(),
             matchCase: this.shouldMatchCase()
         };
         this.resultTreeWidget.search(this.searchTerm, searchOptions);
     }
 
+    protected shouldFollowSymlinks(): boolean {
+        if (this.symlinkState.enabled) {
+            return this.symlinkState.enabled;
+        }
+        return (
+            !!this.searchInWorkspacePreferences['search.followSymlinks']
+        );
+    }
```

Comment:
Update to simply check the preference value:

```suggestion
    protected shouldFollowSymlinks(): boolean {
        return this.searchInWorkspacePreferences['search.followSymlinks']
    }
```